### PR TITLE
feat: add `hexToRgba` util function

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-NNNN]

### Description

New util function to be able to use `Theme.palette` colors even when we need `rgba` format.

### How to test

- check new unit tests to see how it works

### Screenshots

`N/A`

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- N/A Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- N/A Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- N/A Annotate all `props` in component with documentation
- [x] Create `examples` for component
- N/A Ensure that deployed demo has expected results and good examples
- N/A Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- N/A codemod is created and showcased in the changeset
- N/A test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: packages/shared/src/styles/hex-to-rgba.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const hexToRgba = (hexWithHash: string, opacity: number): string => {
+  const hex = hexWithHash.replace('#', '')
```
Comment (dmaklygin) [inline packages/shared/src/styles/hex-to-rgba.ts:2]:
```suggestion
  const hex = hexCode.substring(1)
```

File: packages/shared/src/styles/hex-to-rgba.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+  const red = parseInt(hex.substring(0, 2), 16)
+  const green = parseInt(hex.substring(2, 4), 16)
+  const blue = parseInt(hex.substring(4, 6), 16)
```
Comment (dmaklygin) [inline packages/shared/src/styles/hex-to-rgba.ts:6]:
```suggestion
  const [red, green, blue] = [0, 2, 4].map((startIndex) =>
    parseInt(hex.substring(startIndex, startIndex + 2), 16)
  )
```

wdyt?

**Review by dmaklygin — APPROVED** [review]
Looks good

File: packages/shared/src/styles/hex-to-rgba.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const hexToRgba = (hexWithHash: string, opacity: number): string => {
```
[the remark this answers is not in the document]
Comment (dmaklygin) [reply packages/shared/src/styles/hex-to-rgba.ts:1]: About `hexWithHash`: I would also validate HEX value

File: packages/shared/src/styles/test.ts [hunk 4]
```diff
@@ -39,3 +39,25 @@ describe('rem to px units converter using the BASE font size by default', () =>
     expect(pxFromRem('1rem', 20)).toBe('20px')
   })
 })
+
+describe('hexToRgba', () => {
+  it('should convert #000 with opacity 1 to rgba(0, 0, 0, 1)', () => {
```
Comment (dmaklygin) [inline packages/shared/src/styles/test.ts:44]:
```suggestion
  describe('when opacity 1 and hex is #000', () => {
  it('converts to rgba(0,  0, 0, 1)', () => {
```

File: packages/shared/src/styles/hex-to-rgba.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (TomasSlama) [reply packages/shared/src/styles/hex-to-rgba.ts:1]: 👍

File: packages/shared/src/styles/hex-to-rgba.ts (hunk 2 above)
Comment (TomasSlama) [reply packages/shared/src/styles/hex-to-rgba.ts:24]: To me my solution seems more readable 🤔

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: packages/shared/src/styles/hex-to-rgba.ts (hunk 2 above)
Comment (dmaklygin) [reply packages/shared/src/styles/hex-to-rgba.ts:24]: let's keep yours

**Review by dmaklygin — APPROVED** [review]
Good job!

**Review by mkrl — COMMENTED** [review]
Before this is merged, please take a look at my comment here https://github.com/toptal/picasso/pull/3421/files#r1132209985, perhaps we already have a similar functionality in picasso-shared?

**Comment by TomasSlama** [comment]
@mkrl thank you ❤️, didn't know about this util function 👍

**Pull request closed without merging**